go.mod

```diff
@@ -1,9 +1,84 @@
 module github.com/tsuru/terraform-provider-hcaas
 
-go 1.16
+go 1.18
 
 require (
-	github.com/hashicorp/terraform-plugin-docs v0.5.0 // indirect
-	github.com/hashicorp/terraform-plugin-sdk/v2 v2.8.0
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.23.0
```

Comment:
Would you please consider 2 different commits:
- one for upgrading go
- another for upgrading terraform sdk?

It's easier to maintain and pinpoint any issue in the future.

Reply:
não é possível atualizar o SDK sem atualizar o go é incompatível